src/borg/repository.py

```diff
@@ -182,9 +182,29 @@ def create(self, path):
 
     def save_config(self, path, config):
         config_path = os.path.join(path, 'config')
+        old_config_path = os.path.join(path, 'config.old')
+
+        if os.path.isfile(old_config_path):
+            logger.warning("Old config file not securely erased on previous config update")
+            self.secure_erase_config(old_config_path)
+
+        if os.path.isfile(config_path):
+            os.link(config_path, old_config_path)
```

Comment:
maybe try this on ntfs. from os.link docs it seems that hardlink support was added for windows a while ago. ntfs has junctions, so it might even work. ntfs is quite popular on external usb disks.

but even if it works for ntfs, we could still have a try/except here and track whether making the hardlink worked.
if not, no need to secure erase it later.